qiita_db/processing_job.py

```diff
@@ -488,9 +490,40 @@ def get_resource_allocation_info(self):
                     AssertionError(
                         "Could not match %s to a resource allocation!" % name)
 
-            # [0] sending one element as execute_fetchflatten returns
-            # an array
-            return result[0]
+            allocation = result[0]
+
+            if ('{samples}' in allocation or '{columns}' in allocation or
+                    '{input_size}' in allocation):
+                samples, columns, input_size = self.shape
+                parts = []
+                for part in allocation.split(' '):
+                    if ('{samples}' in part or '{columns}' in part or
+                            '{input_size}' in part):
+                        variable, value = part.split('=')
+                        error_msg = ('Obvious incorrect allocation. Please '
+                                     'contact qiita.help@gmail.com')
+                        # to make sure that the formula is correct and avoid
+                        # possible issues with conversions, we will check that
+                        # all the variables {samples}/{columns}/{input_size}
+                        # present in the formula are not None, if any is None
+                        # we will set the job's error (will stop it) and the
+                        # message is gonna be shown to the user within the job
+                        if (('{samples}' in value and samples is None) or
+                                ('{columns}' in value and columns is None) or
+                                ('{input_size}' in value and input_size is
+                                 None)):
+                            self._set_error(error_msg)
+                            return 'Not valid'
+                        value = naturalsize(eval(value.format(
+                            samples=samples, columns=columns,
+                            input_size=input_size)), gnu=True, format='%.0f')
+
```

Comment:
Comments helped a lot, thanks! Should we check that the allocation is always >0? Or for that matter that the result of eval is a positive integer?

Reply:
adding ...